private/globals_repo.bzl

```
@@ -1,10 +1,29 @@
 load("//private:parse.bzl", "parse_version")
 
 def _globals_repo_impl(rctx):
-    rctx.file("BUILD.bazel")
-
     bazel_version = parse_version(native.bazel_version)
 
+    allow_unresolved_symlinks_flag = (
+        "experimental_allow_unresolved_symlinks"
+        if bazel_version < parse_version("7.0.0")
+        else "allow_unresolved_symlinks"
+    )
+
+    rctx.file("BUILD.bazel", """\
+# @generated by bazel_features/private/globals_repo.bzl
+
+# Allow rules to sense the value of this flag, which was renamed.
+config_setting(
+    name = "allow_unresolved_symlinks",
+    values = {allow_unresolved_symlinks},
+    visibility = ["//visibility:public"],
+)
+""".format(
+    allow_unresolved_symlinks = {
+        allow_unresolved_symlinks_flag: "true"
+    },
```

```suggestion
    allow_unresolved_symlinks = repr({
        allow_unresolved_symlinks_flag: "true"
    }),
```
to make it more obvious that we generate code for the dict.